File: Audiograph/AddToQueue.vb
Imports WMPLib
Public Class frmAddToQueue
    Private Sub Cancel(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub

    Private Sub OK(sender As Object, e As EventArgs) Handles btnOK.Click
        ' if there is nothing in box just close the form
        If txtLocation.Text = String.Empty Then
            Me.Close()
            Exit Sub
        End If

        ' get initial amount of queue items for later
        Dim queueitems As Integer = frmMain.ltvMediaQueue.Items.Count()

        ' get array of files
        Dim names() As String = txtLocation.Text.Split("|"c)
        ' loop through files to check for invalid
        For Each val As String In names
            If val.Contains(":\") = False AndAlso val.Contains("http://") = False AndAlso val.Contains("https://") = False Then
                MessageBox.Show("The location entered does not appear to be a valid filesystem location or URL.", "Invalid Location", MessageBoxButtons.OK, MessageBoxIcon.Error)
                Exit Sub
            End If
        Next

        ' add each element to queue
        For Each val As String In names
            ' only add file name if its a filesystem location, add entire url if its a url
            If val.Contains(":\") = True Then
                ' determine file type
                If val.ToLower.Contains(".mp3") = True OrElse val.ToLower.Contains(".aac") = True OrElse val.ToLower.Contains(".flac") = True OrElse val.ToLower.Contains(".wav") = True OrElse
                    val.ToLower.Contains(".wma") = True OrElse val.ToLower.Contains(".m4a") = True OrElse val.ToLower.Contains(".mid") = True Then    ' audio types
                    frmMain.ltvMediaQueue.Items.Add(" - " & val.Substring(InStrRev(val, "\"))).ImageIndex = 0
                ElseIf val.ToLower.Contains(".mp4") = True OrElse val.ToLower.Contains(".mov") = True OrElse val.ToLower.Contains(".mpeg") = True OrElse val.ToLower.Contains(".mpg") = True OrElse
                    val.ToLower.Contains(".avi") = True OrElse val.ToLower.Contains(".wmv") = True Then ' video types
                    frmMain.ltvMediaQueue.Items.Add(" - " & val.Substring(InStrRev(val, "\"))).ImageIndex = 1
                Else
                    ' anything else
                    frmMain.ltvMediaQueue.Items.Add(" - " & val.Substring(InStrRev(val, "\"))).ImageIndex = 2
                End If
            Else
                ' link
                frmMain.ltvMediaQueue.Items.Add((frmMain.ltvMediaQueue.Items.Count + 1).ToString & " - " & val).ImageIndex = 3
            End If
            ' add subitem with full location
            frmMain.ltvMediaQueue.Items(frmMain.ltvMediaQueue.Items.Count - 1).SubItems.Add(val)
        Next

        ' shuffle crap
        If frmMain.chkMediaShuffle.Checked = True Then
            addingqueue = True
            frmMain.chkMediaShuffle.Checked = False
        End If

        ' begin playing if nothing is in queue or in the player, only recount if not
        If queueitems < 1 AndAlso frmMain.MediaPlayer.playState <> WMPPlayState.wmppsPlaying AndAlso frmMain.MediaPlayer.playState <> WMPPlayState.wmppsPaused Then
            frmMain.QueuePlay(0)
        Else
            frmMain.QueueRecount()
        End If

        ' close form
        Me.Close()
    End Sub

    Private Sub Browse(sender As Object, e As EventArgs) Handles btnBrowse.Click
        OpenFileDialog.ShowDialog()
        Dim names() As String = OpenFileDialog.FileNames
        ' put item(s) in text box separated by quotation marks and a space
        Dim first As Boolean = False

        If names.Length > 1 Then    ' dont run the first time to prevent | from being inserted at the start
            For Each val As String In names
                If first = True Then
                    ' this will run after the first time
                    txtLocation.Text &= "|" & val
                Else
                    ' this will run the first time
                    txtLocation.Text = val
                End If
                first = True
            Next
        ElseIf names.Length = 1 Then
            ' just set the plain text if only one item was selected
            txtLocation.Text = names(0)
        End If
    End Sub
End Class

File: Audiograph/AddToQueue.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using WMPLib;

namespace Audiograph
{
    public partial class frmAddToQueue
    {
        public frmAddToQueue()
        {
            InitializeComponent();
        }
        private void Cancel(object sender, EventArgs e)
        {
            Close();
        }

        private void OK(object sender, EventArgs e)
        {
            // if there is nothing in box just close the form
            if (string.IsNullOrEmpty(txtLocation.Text))
            {
                Close();
                return;
            }

            // get initial amount of queue items for later
            int queueitems = My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Count;

            // get array of files
            string[] names = txtLocation.Text.Split('|');
            // loop through files to check for invalid
            foreach (string val in names)
            {
                if (val.Contains(@":\") == false && val.Contains("http://") == false && val.Contains("https://") == false)
                {
                    MessageBox.Show("The location entered does not appear to be a valid filesystem location or URL.", "Invalid Location", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    return;
                }
            }

            // add each element to queue
            foreach (string val in names)
            {
                // only add file name if its a filesystem location, add entire url if its a url
                if (val.Contains(@":\") == true)
                {
                    // determine file type
                    if (val.ToLower().Contains(".mp3") == true || val.ToLower().Contains(".aac") == true || val.ToLower().Contains(".flac") == true || val.ToLower().Contains(".wav") == true || val.ToLower().Contains(".wma") == true || val.ToLower().Contains(".m4a") == true || val.ToLower().Contains(".mid") == true)    // audio types
                    {
                        My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Add(" - " + val.Substring(Strings.InStrRev(val, @"\"))).ImageIndex = 0;
                    }
                    else if (val.ToLower().Contains(".mp4") == true || val.ToLower().Contains(".mov") == true || val.ToLower().Contains(".mpeg") == true || val.ToLower().Contains(".mpg") == true || val.ToLower().Contains(".avi") == true || val.ToLower().Contains(".wmv") == true) // video types
                    {
                        My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Add(" - " + val.Substring(Strings.InStrRev(val, @"\"))).ImageIndex = 1;
                    }
                    else
                    {
                        // anything else
                        My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Add(" - " + val.Substring(Strings.InStrRev(val, @"\"))).ImageIndex = 2;
                    }
                }
                else
                {
                    // link
                    My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Add((My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Count + 1).ToString() + " - " + val).ImageIndex = 3;
                }
                // add subitem with full location
                My.MyProject.Forms.frmMain.ltvMediaQueue.Items[My.MyProject.Forms.frmMain.ltvMediaQueue.Items.Count - 1].SubItems.Add(val);
            }

            // shuffle crap
            if (My.MyProject.Forms.frmMain.chkMediaShuffle.Checked == true)
            {
                Utilities.addingqueue = true;
                My.MyProject.Forms.frmMain.chkMediaShuffle.Checked = false;
            }

            // begin playing if nothing is in queue or in the player, only recount if not
            if (queueitems < 1 && My.MyProject.Forms.frmMain.MediaPlayer.playState != WMPPlayState.wmppsPlaying && My.MyProject.Forms.frmMain.MediaPlayer.playState != WMPPlayState.wmppsPaused)
            {
                My.MyProject.Forms.frmMain.QueuePlay(0);
            }
            else
            {
                My.MyProject.Forms.frmMain.QueueRecount();
            }

            // close form
            Close();
        }

        private void Browse(object sender, EventArgs e)
        {
            OpenFileDialog.ShowDialog();
            string[] names = OpenFileDialog.FileNames;
            // put item(s) in text box separated by quotation marks and a space
            bool first = false;

            if (names.Length > 1)    // dont run the first time to prevent | from being inserted at the start
            {
                foreach (string val in names)
                {
                    if (first == true)
                    {
                        // this will run after the first time
                        txtLocation.Text += "|" + val;
                    }
                    else
                    {
                        // this will run the first time
                        txtLocation.Text = val;
                    }
                    first = true;
                }
            }
            else if (names.Length == 1)
            {
                // just set the plain text if only one item was selected
                txtLocation.Text = names[0];
            }
        }
    }
}